### PR TITLE
Adding set_value method to GeneralVarLikeExpression

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
@dallan-keylogic identified an issue where using `set_value()` on the IDAES `GeneralVarLikeExpression` component would overwrite the expression with the new value, which was undesired behaviour. This PR overloads `set_value` to raise an `Exception` in normal use, but retains a `force` option to allow users to change the expression if desired.

## Changes proposed in this PR:
- Overloads `set_value()`
- Adds new tests for `set_value`
- Fixes a case where `value()` should have been used in a test.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/core/util/misc.py [hunk 1]
```diff
@@ -254,14 +254,39 @@ class _GeneralVarLikeExpressionData(_GeneralExpressionData):
...
+                f"{self.name} is an Expression and does not have a value "
+                f"which can be set.")
+
     @property
     def value(self):
-        # Overload value so it behaves like a Var
-        return pyo.value(self.expr)
+        raise TypeError(
+            f"{self.name} is an Expression and does not have a value "
+            f"attribute. Use the 'value()' method instead.")
```
Comment (jsiirola) [inline idaes/core/util/misc.py:286]: I like this: `.value` on `Expression` was deprecated a long time ago...  Given #504, I think making ths louder is a good thing.

**Review by jsiirola — APPROVED** [review]
Looks good, pending successful tests.

File: idaes/core/util/misc.py (hunk 1 above)
Comment (andrewlee94) [reply idaes/core/util/misc.py:286]: That was the reason I maintained it.

**Review by eslickj — APPROVED** [review]
This looks fine as far as I can tell, but I have a few questions.

1) Should simple be scalar?
2) Isn't this a problem that could hit Pyomo too?  I mean for simple model or models you are very familiar with it's probably not an issue.  But for complex models you didn't write or modular models where there is no guarantee that a particular quantity is an expression or variable, won't there be a high risk of user error?
3) Should I not be using Pyomo Expressions (because I use them a lot probably going around this check)?

**Comment by andrewlee94** [comment]
@eslickj WE deliberately made this IDAES-specific for the particular case of the outlet of Separators (i.e. where we cheat an populate the Ports with Expressions rather than Vars).

So:

1. We could make the transition to Scalar, but I am not sure how much it matters.
2. As I said, we made this IDAES specific for a reason. Pyomo didn't want to make Expressions look like Vars, and they have a good use for `set_Value` in `Expressions`.
3. No, you should use Pyomo `Expressions` in most cases. These are intended for very specific cases where there is a high likelihood of a user confusing the `Expression` for a `Var`.

**Pull request merged**